### PR TITLE
Integrate Login Page And Add Relevant Redirections

### PR DESCRIPTION
**Pull request by mahadzaryab1**
### Notion ticket link

[Log in integration](https://www.notion.so/uwblueprintexecs/Log-in-integration-43780f8ddca34491befa6bdb1235a68b)

### Implementation description
- Integrated the login page that was built with NextAuth.js

### Steps to test

~~1. Heading over to http://localhost:3000/ should redirect you to the login page if you're not already authenticated~~
2. If you're authenticated, then you'll be redirected to the root page with a message that you're authenticated

### Checklist

-   [x] My PR name is descriptive and in imperative tense
-   [x] I have run the linter
-   [x] I have requested a review from the PL, and/or other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Comment by jason17h** [comment]
Quick question - what is the expected behaviour for after I enter my email and click "Continue"? It currently says "An unexpected error occurred". Is that being implemented later or was that meant to be handled in this PR?

File: pages/index.tsx [hunk 1]
```diff
@@ -17,3 +23,27 @@ export default function Component() {
...
+
+/**
+ * getServerSideProps runs before each page is rendered to check to see if a
+ * user has already been authenticated
+ */
+export async function getServerSideProps(context: GetSessionOptions) {
+    // obtain the next auth session
+    const session = await getSession(context);
+
+    // if the user is not authenticated redirect them to the login page
```
Comment (feng-eric) [inline pages/index.tsx]: Should we be redirecting them to the login page or the home page?

**Review by feng-eric — COMMENTED** [review]
Had one small question.

**Comment by feng-eric** [comment]
> Quick question - what is the expected behaviour for after I enter my email and click "Continue"? It currently says "An unexpected error occurred". Is that being implemented later or was that meant to be handled in this PR?

I think @victoranyun is going to be working on this.

File: pages/index.tsx (hunk 1 above)
Comment (jason17h) [reply pages/index.tsx]: That's what I was wondering at first as well, but if our platform is specifically for the class management stuff then I would assume you need to log in to be able to access anything? Or are you allowed to browse programs even if you're not registered?

Comment (feng-eric) [reply pages/index.tsx]: You're allowed to browse programs even if you're not logged in.

Comment (feng-eric) [reply pages/index.tsx]: It's something we discussed in May/June during our team meetings

File: pages/login.tsx [hunk 2]
```diff
@@ -6,12 +6,22 @@ import {
...
  * This is the page that a user will use to either login or register
  * to the SDC platform
  */
 export default function Login(): JSX.Element {
+    // hook to hold the user's email
+    const [email, setEmail] = useState("");
+
+    // signInWithEmail sends a login request to the user's email
+    const signInWithEmail = () => {
+        signIn("email", { email: email });
```
Comment (feng-eric) [inline pages/login.tsx]: Thought's on making "email" a const?

Comment (feng-eric) [reply pages/login.tsx]: Also, I think it can just be:
```suggestion
        signIn("email", { email });
```
Let me know if this works.

**Review by feng-eric — APPROVED** [review]

**Pull request merged**